executor/join.go

```diff
@@ -325,35 +329,37 @@ func (e *HashJoinExec) partitionInnerRows() error {
 		return err
 	}
 
-	wg := &sync.WaitGroup{}
-	defer wg.Wait()
-	wg.Add(int(e.concurrency))
+	e.partWorkerWaitGroup = sync.WaitGroup{}
```

Comment:
I think we don't need to make it to be a member field of `HashJoinExec`. Why not keep the old way?
```go
go util.WithRecovery(func() {
	defer wg.Done()
	e.doInnerPartition(workerID)
}, e.finishPartitionInnerRows)
```

Reply:
`wg.Done` is moved to `finishPartitionInnerRows`